File: 2299-merge-nodes-in-between-zeros/2299-merge-nodes-in-between-zeros.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeNodes(ListNode head) {
       
        if(head == null || head.next == null){
            return head;
        }

        ListNode prev = head;
        ListNode curr = head.next;

        while(curr != null){    
            int sum = 0;

            while(curr.val != 0){
                sum += curr.val;
                curr = curr.next;
            }

            prev.val = sum;
            prev.next = (curr.next == null) ? null : curr;
            prev = curr;
            curr = curr.next;
        } 

        return head;
    }
}

File: 2299-merge-nodes-in-between-zeros/2299-merge-nodes-in-between-zeros.py
class Solution:
    def mergeNodes(self, head):
        if not head or not head.next:
            return head

        prev = head
        curr = head.next

        while curr:
            curr_sum = 0
            while curr.val != 0:
                curr_sum += curr.val
                curr = curr.next

            prev.val = curr_sum
            prev.next = None if not curr.next else curr
            prev = curr
            curr = curr.next

        return head
